### ISSUE TITLE
Manual deletion of S3 bucket CORS does not force terraform to update CORS for bucket.

### ISSUE THREAD

**Issue by jessegonzalez** [terraform-providers/terraform-provider-aws#4849]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
```
Terraform v0.11.7
+ provider.aws v1.23.0
```

### Affected Resource(s)

* aws_s3_bucket

### Terraform Configuration Files

```hcl
provider "aws" {
  region  = "us-east-1"
  version = " ~> 1.23.0"
  profile = "dev"
}

resource "aws_s3_bucket" "cors-testing-terraform" {
  bucket = "cors-testing-terraform"

  cors_rule {
    allowed_origins = ["some.fqdn", "another.fqdn"]
    allowed_methods = ["GET"]
    allowed_headers = ["*"]
    max_age_seconds = 3000
  }
}
```

### Expected Behavior

After having deleted CORS configuration for an S3 bucket in the AWS console, terraform plan/apply should recognize the change and apply CORS configuration as defined.

### Actual Behavior

CORS is left unconfigured.

### Steps to Reproduce

1. `terraform apply`
2. Update S3 bucket and delete CORS configuration
3. `terraform apply`

### Important Factoids

Other changes to the CORS configuration via console , i.e. adding AllowedOrigin or AllowHeader attributes are recognized and adjusted.

### References

* None

**Comment by jessegonzalez** [comment]
I've created a PR for this issue: #4887

**Comment by bflad** [comment]
The fix for this has been merged into master and will release with version 1.24.0 of the AWS provider, very shortly.

**Comment by bflad** [comment]
This has been released in [version 1.24.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1240-june-21-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
